### PR TITLE
Show uses of raw palette indices

### PR DESCRIPTION
**Pull request by marijnvdwerf**

### CODE REVIEW

File: src/interface/widget.c [hunk 1]
```diff
@@ -1116,6 +1116,7 @@ static void widget_text_box_draw(rct_drawpixelinfo *dpi, rct_window *w, int widg
 
 	if (gTextBoxFrameNo <= 15){
 		uint8 colour = ColourMapA[w->colours[1]].mid_light;
+		// TODO: Colour addition
 		gfx_fill_rect(dpi, cur_x, t + 9, cur_x + width, t + 9, colour + 5);
```
Comment (marijnvdwerf) [inline src/interface/widget.c]: `ColourMapA[w->colours[1]].colour_11`?

File: src/interface/console.c [hunk 2]
```diff
@@ -236,13 +236,13 @@ void console_draw(rct_drawpixelinfo *dpi)
 		int caretX = x + gfx_get_string_width(lineBuffer);
 		int caretY = y + lineHeight;
 
-		gfx_fill_rect(dpi, caretX, caretY, caretX + 6, caretY + 1, FORMAT_WHITE);
+		gfx_fill_rect(dpi, caretX, caretY, caretX + 6, caretY + 1, PALETTE_INDEX_144);
 	}
-	gfx_fill_rect(dpi, _consoleLeft, _consoleBottom - 21, _consoleRight, _consoleBottom - 21, 14);
-	gfx_fill_rect(dpi, _consoleLeft, _consoleBottom - 20, _consoleRight, _consoleBottom - 20, 11);
+	gfx_fill_rect(dpi, _consoleLeft, _consoleBottom - 21, _consoleRight, _consoleBottom - 21, PALETTE_INDEX_14);
+	gfx_fill_rect(dpi, _consoleLeft, _consoleBottom - 20, _consoleRight, _consoleBottom - 20, PALETTE_INDEX_11);
```
Comment (marijnvdwerf) [inline src/interface/console.c]:
```
shadow	= ColourMapA[COLOUR_BLACK].mid_dark;
hilight	= ColourMapA[COLOUR_BLACK].lighter;
```
would probably make sense (matching `gfx_fill_rect_inset`)

**Comment by Gymnasiast** [comment]
What's your plan with this, @marijnvdwerf ?

**Comment by marijnvdwerf** [comment]
The palette indexese should be replaced by something like `ColourMapA[COLOUR_RED].lightest` (or a new window colour), but that needs to be done manually because some palette indexes are used in more than one colour scheme: http://jsbin.com/kesade/2/edit?html,output

**Comment by Gymnasiast** [comment]
Should this be merged, or?

**Comment by marijnvdwerf** [comment]
I was originally planning to have this as a replacement for a `// TODO:` everywhere. But we might as well just merge it, at least the enums are more obvious than bare numbers.

**Pull request merged**